Yoast/wordpress-seo
# DUPP-27 update workouts overview free

## Context


The current workout overview in free consists of half the cornerstone workout that ends in an upsell. It needs to be replaced with an overview that shows all the existing workouts. The free workouts will be accessible to all users. The premium workouts will look different and feature an “unlock with premium” button instead of a “Start workout” button. See the [design](https://www.sketch.com/s/5c4b9671-7370-41f7-91ff-c923c43e5859/a/1KmxlWn/play). 

The unlock with premium buttons should have the following shortlinks
* cornerstone: https://yoa.st/workout-cornerstone-upsell
* orphaned: https://yoa.st/workout-orphaned-content-upsell

This PR implements the new design for the free and premium cards and removes the upsell modal and introduces the upsell buttons.

## Summary


This PR can be summarized in the following changelog entry:

* Updates the design of the workout cards.

## Relevant technical choices:

* 

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

_Free_
* Checkout this branch and build.
* Make sure premium is disabled.
* Navigate to SEO > Workouts.
* See it matches the design and check the upsell buttons go to the corresponding upsell pages.
<img width="549" alt="Screenshot 2021-11-10 at 10 46 26" src="https://user-images.githubusercontent.com/43582255/141094371-87397f4f-f505-48aa-b8bb-22e8c3a3e037.png">

_Premium_
* In `workouts-integration.php` set 17.7 to 17.5 or lower in the `should_update_premium` method.
* Activate premium and refresh the workouts page.
* See it matches the design and check the buttons open the corresponding workouts.
<img width="537" alt="Screenshot 2021-11-10 at 10 44 46" src="https://user-images.githubusercontent.com/43582255/141094474-a013cb2a-ddd8-4655-8842-43bea35a49b9.png">

### Test instructions for QA when the code is in the RC


* [] QA should use the same steps as above.


QA can test this PR by following these steps:

* Make sure premium is disabled.
* Navigate to SEO > Workouts.
* See it matches the design and check the upsell buttons go to the corresponding upsell pages.
* Activate premium (the latest RC) and refresh the workouts page.
* See it matches the design and check the buttons open the corresponding workouts.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes [DUPP-27]


[DUPP-27]: https://yoast.atlassian.net/browse/DUPP-27?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
